### PR TITLE
feat: handle proxy port link, flows network convergence

### LINKED ISSUE #28: feat: subscribe to `kytos/topology.link_up|link_down` to handle proxy port (looped) up/down
For more details, check out the augmented blueprint on https://github.com/kytos-ng/kytos/pull/402/

Whether or not the telemetry EVC will remove the flows when a link of a proxy port is down is still being discussed, probably it will.

### PR DESCRIPTION
**Pull request by viniarck**
Closes #28 

### Summary

- Initial PR supporting covergence for a (looped) link used as a proxy port
- started handling proxy port link_up|link_down events
- started handling mef_eline loaded event
- extracted a few methods to reuse flow installations/deletions
- initial dict structures to keep track of unis sources and proxy ports for subsequent PRs 

### Local Tests

I tested out on INT lab the following cases with 2 inter EVPLs (sw1 and sw6) and one intra EVPL (on sw1):

- I also performed restarts to make sure it loaded correctly 
- I shut down a loop and observed that only the expected evcs got their flows removed falling back to mef_eline flows, I also had iperf3 running to make sure traffic would still flow with a temporary drop:

```
kytos $> 2023-11-13 12:41:10,036 - INFO [kytos.napps.kytos/of_core] [main.py:701:update_port_status] (MainThread) PortStatus modified interface 00:00:00:00:00:00:00:01:17 state OFPPS_LI
NK_DOWN
2023-11-13 12:41:10,041 - INFO [kytos.napps.kytos/mef_eline] [main.py:764:handle_link_down] (thread_pool_app_15) Event handle_link_down Link(Interface('novi_port_17', 17, Switch('00:00:
00:00:00:00:00:01')), Interface('novi_port_18', 18, Switch('00:00:00:00:00:00:00:01')), f8a7b8b87861d1dca6a471610d8b0c33ce320d097d41f27286f65391a5d60cd2)
2023-11-13 12:41:10,061 - INFO [kytos.napps.kytos/of_core] [main.py:701:update_port_status] (MainThread) PortStatus modified interface 00:00:00:00:00:00:00:01:18 state OFPPS_LINK_DOWN
2023-11-13 12:41:10,067 - INFO [uvicorn.access] [httptools_impl.py:506:send] (MainThread) 127.0.0.1:56238 - "GET /api/kytos/mef_eline/v2/evc/?archived=false&metadata.telemetry.enabled=t
rue&metadata.telemetry.status=UP HTTP/1.1" 200
2023-11-13 12:41:10,067 - INFO [kytos.napps.kytos/telemetry_int] [int.py:102:handle_pp_link_down] (MainThread) Handling link_down Link(Interface('novi_port_17', 17, Switch('00:00:00:00:
00:00:00:01')), Interface('novi_port_18', 18, Switch('00:00:00:00:00:00:00:01')), f8a7b8b87861d1dca6a471610d8b0c33ce320d097d41f27286f65391a5d60cd2), removing INT flows falling back to m
ef_eline, EVC ids: ['818a626a14474b', 'ede8d068444443', 'f7dc68ef737d46']
2023-11-13 12:41:10,078 - INFO [uvicorn.access] [httptools_impl.py:506:send] (MainThread) 127.0.0.1:56252 - "GET /api/kytos/flow_manager/v2/stored_flows?state=installed&state=pending&co
okie_range=12142138225658709835&cookie_range=12142138225658709835&cookie_range=12172641349652464707&cookie_range=12172641349652464707&cookie_range=12175442460871458118&cookie_range=1217
5442460871458118 HTTP/1.1" 200
2023-11-13 12:41:10,085 - INFO [kytos.napps.kytos/flow_manager] [utils.py:196:flows_to_log_info] (thread_pool_app_25) Send FlowMod from KytosEvent dpid: 00:00:00:00:00:00:00:06, command: delete, force: True,  flows[0, 2]: [{'cookie': 12142138225658709835, 'cookie_mask': 18446744073709551615, 'table_id': 255}, {'cookie': 12175442460871458118, 'cookie_mask': 18446744073709551615, 'table_id': 255}]
2023-11-13 12:41:10,087 - INFO [kytos.napps.kytos/flow_manager] [utils.py:196:flows_to_log_info] (thread_pool_app_13) Send FlowMod from KytosEvent dpid: 00:00:00:00:00:00:00:01, command: delete, force: True,  flows[0, 3]: [{'cookie': 12142138225658709835, 'cookie_mask': 18446744073709551615, 'table_id': 255}, {'cookie': 12172641349652464707, 'cookie_mask': 18446744073709551615, 'table_id': 255}, {'cookie': 12175442460871458118, 'cookie_mask':18446744073709551615, 'table_id': 255}]
2023-11-13 12:41:10,112 - INFO [uvicorn.access] [httptools_impl.py:506:send] (MainThread) 127.0.0.1:56264 - "POST /api/kytos/mef_eline/v2/evc/metadata HTTP/1.1" 201
2023-11-13 12:41:12,102 - INFO [kytos.napps.kytos/of_lldp] [loop_manager.py:250:handle_loop_stopped] (thread_pool_app_20) LLDP loop stopped on switch: 00:00:00:00:00:00:00:01, interfaces: ['novi_port_17', 'novi_port_18'], port_numbers: [17, 18]
2023-11-13 12:41:12,115 - INFO [uvicorn.access] [httptools_impl.py:506:send] (MainThread) 127.0.0.1:56272 - "DELETE /api/kytos/topology/v3/interfaces/00%3A00%3A00%3A00%3A00%3A00%3A00%3A
01%3A17/metadata/looped HTTP/1.1" 200
kytos $>                                                                                                                                                                                 

```

```
root@int02:/home/amlight# iperf3 -c 10.22.21.3 -i 5 -t 60
Connecting to host 10.22.21.3, port 5201
[  4] local 10.22.21.2 port 56434 connected to 10.22.21.3 port 5201
[ ID] Interval           Transfer     Bandwidth       Retr  Cwnd
[  4]   0.00-5.00   sec  11.1 GBytes  19.0 Gbits/sec  1323    624 KBytes       
[  4]   5.00-10.00  sec  12.2 GBytes  20.9 Gbits/sec  778    960 KBytes       
[  4]  10.00-15.00  sec  8.54 GBytes  14.7 Gbits/sec  562    751 KBytes       
[  4]  15.00-20.00  sec  9.96 GBytes  17.1 Gbits/sec  428    634 KBytes       
[  4]  20.00-25.00  sec  11.7 GBytes  20.1 Gbits/sec  754    793 KBytes       
[  4]  25.00-30.00  sec  11.7 GBytes  20.0 Gbits/sec  646    909 KBytes       
[  4]  30.00-35.00  sec  12.3 GBytes  21.1 Gbits/sec  853    875 KBytes       
[  4]  35.00-40.00  sec  10.4 GBytes  17.9 Gbits/sec  753    492 KBytes       
[  4]  40.00-45.00  sec  10.0 GBytes  17.2 Gbits/sec  1061    884 KBytes       
[  4]  45.00-50.00  sec  9.51 GBytes  16.3 Gbits/sec  409    650 KBytes       
[  4]  50.00-55.00  sec  12.6 GBytes  21.7 Gbits/sec  956    571 KBytes       
[  4]  55.00-60.00  sec  12.0 GBytes  20.7 Gbits/sec  638    773 KBytes       
- - - - - - - - - - - - - - - - - - - - - - - - -
[ ID] Interval           Transfer     Bandwidth       Retr
[  4]   0.00-60.00  sec   132 GBytes  18.9 Gbits/sec  9161             sender
[  4]   0.00-60.00  sec   132 GBytes  18.9 Gbits/sec                  receiver

iperf Done.
```

- I brought the loop up again and observed that it deployed again:

```
kytos $> 2023-11-13 12:41:35,266 - INFO [kytos.napps.kytos/of_core] [main.py:701:update_port_status] (MainThread) PortStatus modified interface 00:00:00:00:00:00:00:01:18 state OFPPS_LI
VE
2023-11-13 12:41:35,500 - INFO [kytos.napps.kytos/of_core] [main.py:701:update_port_status] (MainThread) PortStatus modified interface 00:00:00:00:00:00:00:01:17 state OFPPS_LIVE
2023-11-13 12:41:36,452 - INFO [uvicorn.access] [httptools_impl.py:506:send] (MainThread) 127.0.0.1:51852 - "POST /api/kytos/topology/v3/interfaces/00%3A00%3A00%3A00%3A00%3A00%3A00%3A01
%3A17/metadata HTTP/1.1" 201
2023-11-13 12:41:37,511 - INFO [kytos.napps.kytos/mef_eline] [main.py:729:handle_link_up] (thread_pool_app_20) Event handle_link_up Link(Interface('novi_port_17', 17, Switch('00:00:00:0
0:00:00:00:01')), Interface('novi_port_18', 18, Switch('00:00:00:00:00:00:00:01')), f8a7b8b87861d1dca6a471610d8b0c33ce320d097d41f27286f65391a5d60cd2)
2023-11-13 12:41:37,519 - INFO [uvicorn.access] [httptools_impl.py:506:send] (MainThread) 127.0.0.1:51866 - "GET /api/kytos/topology/v3/links HTTP/1.1" 200
2023-11-13 12:41:37,527 - INFO [uvicorn.access] [httptools_impl.py:506:send] (MainThread) 127.0.0.1:51876 - "GET /api/kytos/topology/v3/links HTTP/1.1" 200
2023-11-13 12:41:42,522 - INFO [uvicorn.access] [httptools_impl.py:506:send] (MainThread) 127.0.0.1:51892 - "GET /api/kytos/mef_eline/v2/evc/?archived=false&metadata.telemetry.enabled=t
rue&metadata.telemetry.status=DOWN HTTP/1.1" 200
2023-11-13 12:41:42,523 - INFO [kytos.napps.kytos/telemetry_int] [int.py:177:handle_pp_link_up] (MainThread) Handling link_up Link(Interface('novi_port_17', 17, Switch('00:00:00:00:00:0
0:00:01')), Interface('novi_port_18', 18, Switch('00:00:00:00:00:00:00:01')), f8a7b8b87861d1dca6a471610d8b0c33ce320d097d41f27286f65391a5d60cd2), deploying EVC ids: ['818a626a14474b', 'e
de8d068444443', 'f7dc68ef737d46']
2023-11-13 12:41:42,533 - INFO [uvicorn.access] [httptools_impl.py:506:send] (MainThread) 127.0.0.1:51894 - "GET /api/kytos/flow_manager/v2/stored_flows?state=installed&state=pending HT
TP/1.1" 200
2023-11-13 12:41:42,544 - INFO [kytos.napps.kytos/flow_manager] [utils.py:196:flows_to_log_info] (thread_pool_app_25) Send FlowMod from KytosEvent dpid: 00:00:00:00:00:00:00:01, command
: add, force: True,  flows[0, 24]: [{'owner': 'telemetry_int', 'cookie': 12142138225658709835, 'match': {'in_port': 15, 'dl_vlan': 2221, 'dl_type': 2048, 'nw_proto': 6}, 'table_id': 0, 
'table_group': 'evpl', 'priority': 20100, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_type': 'apply_actions', 'actions': [{'action_type': 'push_int'}]}, {'instr
uction_type': 'goto_table', 'table_id': 2}]}, {'owner': 'telemetry_int', 'cookie': 12142138225658709835, 'match': {'in_port': 15, 'dl_vlan': 2221, 'dl_type': 2048, 'nw_proto': 17}, 'tab
le_id': 0, 'table_group': 'evpl', 'priority': 20100, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_type': 'apply_actions', 'actions': [{'action_type': 'push_int'}
]}, {'instruction_type': 'goto_table', 'table_id': 2}]}, {'owner': 'telemetry_int', 'cookie': 12142138225658709835, 'match': {'in_port': 15, 'dl_vlan': 2221}, 'table_id': 2, 'table_grou
p': 'evpl', 'priority': 20000, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_type': 'apply_actions', 'actions': [{'action_type': 'add_int_metadata'}, {'action_typ
e': 'push_vlan', 'tag_type': 's'}, {'action_type': 'set_vlan', 'vlan_id': 1}, {'action_type': 'output', 'port': 11}]}]}, {'owner': 'telemetry_int', 'cookie': 12142138225658709835, 'matc
h': {'in_port': 11, 'dl_vlan': 1, 'dl_type': 2048, 'nw_proto': 6}, 'table_id': 0, 'table_group': 'evpl', 'priority': 20100, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'inst
ruction_type': 'apply_actions', 'actions': [{'action_type': 'add_int_metadata'}, {'action_type': 'output', 'port': 17}]}]}, {'owner': 'telemetry_int', 'cookie': 12142138225658709835, 'm
atch': {'in_port': 11, 'dl_vlan': 1, 'dl_type': 2048, 'nw_proto': 17}, 'table_id': 0, 'table_group': 'evpl', 'priority': 20100, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'
instruction_type': 'apply_actions', 'actions': [{'action_type': 'add_int_metadata'}, {'action_type': 'output', 'port': 17}]}]}, {'owner': 'telemetry_int', 'cookie': 12142138225658709835
, 'match': {'in_port': 18, 'dl_vlan': 1}, 'table_id': 0, 'table_group': 'evpl', 'priority': 20000, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_type': 'apply_act
ions', 'actions': [{'action_type': 'send_report'}]}, {'instruction_type': 'goto_table', 'table_id': 2}]}, {'owner': 'telemetry_int', 'cookie': 12142138225658709835, 'match': {'in_port':
 18, 'dl_vlan': 1}, 'table_id': 2, 'table_group': 'evpl', 'priority': 20000, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_type': 'apply_actions', 'actions': [{'a
ction_type': 'pop_int'}, {'action_type': 'pop_vlan'}, {'action_type': 'output', 'port': 15}]}]}, {'owner': 'telemetry_int', 'cookie': 12172641349652464707, 'match': {'in_port': 15, 'dl_
vlan': 100, 'dl_type': 2048, 'nw_proto': 6}, 'table_id': 0, 'table_group': 'evpl', 'priority': 20100, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_type': 'apply_
actions', 'actions': [{'action_type': 'push_int'}]}, {'instruction_type': 'goto_table', 'table_id': 2}]}, {'owner': 'telemetry_int', 'cookie': 12172641349652464707, 'match': {'in_port':
 15, 'dl_vlan': 100, 'dl_type': 2048, 'nw_proto': 17}, 'table_id': 0, 'table_group': 'evpl', 'priority': 20100, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_type
': 'apply_actions', 'actions': [{'action_type': 'push_int'}]}, {'instruction_type': 'goto_table', 'table_id': 2}]}, {'owner': 'telemetry_int', 'cookie': 12172641349652464707, 'match': {
'in_port': 15, 'dl_vlan': 100}, 'table_id': 2, 'table_group': 'evpl', 'priority': 20000, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_type': 'apply_actions', 'ac
tions': [{'action_type': 'add_int_metadata'}, {'action_type': 'output', 'port': 19}]}]}, {'owner': 'telemetry_int', 'cookie': 12172641349652464707, 'match': {'in_port': 16, 'dl_vlan': 1
00, 'dl_type': 2048, 'nw_proto': 6}, 'table_id': 0, 'table_group': 'evpl', 'priority': 20100, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_type': 'apply_actions'
, 'actions': [{'action_type': 'push_int'}]}, {'instruction_type': 'goto_table', 'table_id': 2}]}, {'owner': 'telemetry_int', 'cookie': 12172641349652464707, 'match': {'in_port': 16, 'dl
_vlan': 100, 'dl_type': 2048, 'nw_proto': 17}, 'table_id': 0, 'table_group': 'evpl', 'priority': 20100, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_type': 'appl
y_actions', 'actions': [{'action_type': 'push_int'}]}, {'instruction_type': 'goto_table', 'table_id': 2}]}, {'owner': 'telemetry_int', 'cookie': 12172641349652464707, 'match': {'in_port
': 16, 'dl_vlan': 100}, 'table_id': 2, 'table_group': 'evpl', 'priority': 20000, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_type': 'apply_actions', 'actions': 
[{'action_type': 'add_int_metadata'}, {'action_type': 'output', 'port': 17}]}]}, {'owner': 'telemetry_int', 'cookie': 12172641349652464707, 'match': {'in_port': 20, 'dl_vlan': 100}, 'ta
ble_id': 0, 'table_group': 'evpl', 'priority': 20000, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_type': 'apply_actions', 'actions': [{'action_type': 'send_repo
rt'}]}, {'instruction_type': 'goto_table', 'table_id': 2}]}, {'owner': 'telemetry_int', 'cookie': 12172641349652464707, 'match': {'in_port': 20, 'dl_vlan': 100}, 'table_id': 2, 'table_g
roup': 'evpl', 'priority': 20000, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_type': 'apply_actions', 'actions': [{'action_type': 'pop_int'}, {'action_type': 's
et_vlan', 'vlan_id': 100}, {'action_type': 'output', 'port': 16}]}]}, {'owner': 'telemetry_int', 'cookie': 12172641349652464707, 'match': {'in_port': 18, 'dl_vlan': 100}, 'table_id': 0,
 'table_group': 'evpl', 'priority': 20000, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_type': 'apply_actions', 'actions': [{'action_type': 'send_report'}]}, {'i
nstruction_type': 'goto_table', 'table_id': 2}]}, {'owner': 'telemetry_int', 'cookie': 12172641349652464707, 'match': {'in_port': 18, 'dl_vlan': 100}, 'table_id': 2, 'table_group': 'evp
l', 'priority': 20000, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_type': 'apply_actions', 'actions': [{'action_type': 'pop_int'}, {'action_type': 'set_vlan', '
vlan_id': 100}, {'action_type': 'output', 'port': 15}]}]}, {'owner': 'telemetry_int', 'cookie': 12175442460871458118, 'match': {'in_port': 15, 'dl_vlan': 2222, 'dl_type': 2048, 'nw_prot
o': 6}, 'table_id': 0, 'table_group': 'evpl', 'priority': 20100, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_type': 'apply_actions', 'actions': [{'action_type':
 'push_int'}]}, {'instruction_type': 'goto_table', 'table_id': 2}]}, {'owner': 'telemetry_int', 'cookie': 12175442460871458118, 'match': {'in_port': 15, 'dl_vlan': 2222, 'dl_type': 2048
, 'nw_proto': 17}, 'table_id': 0, 'table_group': 'evpl', 'priority': 20100, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_type': 'apply_actions', 'actions': [{'ac
tion_type': 'push_int'}]}, {'instruction_type': 'goto_table', 'table_id': 2}]}, {'owner': 'telemetry_int', 'cookie': 12175442460871458118, 'match': {'in_port': 15, 'dl_vlan': 2222}, 'ta
ble_id': 2, 'table_group': 'evpl', 'priority': 20000, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_type': 'apply_actions', 'actions': [{'action_type': 'add_int_m
etadata'}, {'action_type': 'push_vlan', 'tag_type': 's'}, {'action_type': 'set_vlan', 'vlan_id': 2}, {'action_type': 'output', 'port': 11}]}]}, {'owner': 'telemetry_int', 'cookie': 1217
5442460871458118, 'match': {'in_port': 11, 'dl_vlan': 2, 'dl_type': 2048, 'nw_proto': 6}, 'table_id': 0, 'table_group': 'evpl', 'priority': 20100, 'idle_timeout': 0, 'hard_timeout': 0, 
'instructions': [{'instruction_type': 'apply_actions', 'actions': [{'action_type': 'add_int_metadata'}, {'action_type': 'output', 'port': 17}]}]}, {'owner': 'telemetry_int', 'cookie': 1
2175442460871458118, 'match': {'in_port': 11, 'dl_vlan': 2, 'dl_type': 2048, 'nw_proto': 17}, 'table_id': 0, 'table_group': 'evpl', 'priority': 20100, 'idle_timeout': 0, 'hard_timeout':
 0, 'instructions': [{'instruction_type': 'apply_actions', 'actions': [{'action_type': 'add_int_metadata'}, {'action_type': 'output', 'port': 17}]}]}, {'owner': 'telemetry_int', 'cookie
': 12175442460871458118, 'match': {'in_port': 18, 'dl_vlan': 2}, 'table_id': 0, 'table_group': 'evpl', 'priority': 20000, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instru
ction_type': 'apply_actions', 'actions': [{'action_type': 'send_report'}]}, {'instruction_type': 'goto_table', 'table_id': 2}]}, {'owner': 'telemetry_int', 'cookie': 1217544246087145811
8, 'match': {'in_port': 18, 'dl_vlan': 2}, 'table_id': 2, 'table_group': 'evpl', 'priority': 20000, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_type': 'apply_ac
tions', 'actions': [{'action_type': 'pop_int'}, {'action_type': 'pop_vlan'}, {'action_type': 'output', 'port': 15}]}]}]
2023-11-13 12:41:42,567 - INFO [kytos.napps.kytos/flow_manager] [utils.py:196:flows_to_log_info] (thread_pool_app_13) Send FlowMod from KytosEvent dpid: 00:00:00:00:00:00:00:06, command
: add, force: True,  flows[0, 14]: [{'owner': 'telemetry_int', 'cookie': 12142138225658709835, 'match': {'in_port': 22, 'dl_vlan': 2221, 'dl_type': 2048, 'nw_proto': 6}, 'table_id': 0, 
'table_group': 'evpl', 'priority': 20100, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_type': 'apply_actions', 'actions': [{'action_type': 'push_int'}]}, {'instr
uction_type': 'goto_table', 'table_id': 2}]}, {'owner': 'telemetry_int', 'cookie': 12142138225658709835, 'match': {'in_port': 22, 'dl_vlan': 2221, 'dl_type': 2048, 'nw_proto': 17}, 'tab
le_id': 0, 'table_group': 'evpl', 'priority': 20100, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_type': 'apply_actions', 'actions': [{'action_type': 'push_int'}
]}, {'instruction_type': 'goto_table', 'table_id': 2}]}, {'owner': 'telemetry_int', 'cookie': 12142138225658709835, 'match': {'in_port': 22, 'dl_vlan': 2221}, 'table_id': 2, 'table_grou
p': 'evpl', 'priority': 20000, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_type': 'apply_actions', 'actions': [{'action_type': 'add_int_metadata'}, {'action_typ
e': 'push_vlan', 'tag_type': 's'}, {'action_type': 'set_vlan', 'vlan_id': 1}, {'action_type': 'output', 'port': 11}]}]}, {'owner': 'telemetry_int', 'cookie': 12142138225658709835, 'matc
h': {'in_port': 11, 'dl_vlan': 1, 'dl_type': 2048, 'nw_proto': 6}, 'table_id': 0, 'table_group': 'evpl', 'priority': 20100, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'inst
ruction_type': 'apply_actions', 'actions': [{'action_type': 'add_int_metadata'}, {'action_type': 'output', 'port': 25}]}]}, {'owner': 'telemetry_int', 'cookie': 12142138225658709835, 'm
atch': {'in_port': 11, 'dl_vlan': 1, 'dl_type': 2048, 'nw_proto': 17}, 'table_id': 0, 'table_group': 'evpl', 'priority': 20100, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'
instruction_type': 'apply_actions', 'actions': [{'action_type': 'add_int_metadata'}, {'action_type': 'output', 'port': 25}]}]}, {'owner': 'telemetry_int', 'cookie': 12142138225658709835
, 'match': {'in_port': 26, 'dl_vlan': 1}, 'table_id': 0, 'table_group': 'evpl', 'priority': 20000, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_type': 'apply_act
ions', 'actions': [{'action_type': 'send_report'}]}, {'instruction_type': 'goto_table', 'table_id': 2}]}, {'owner': 'telemetry_int', 'cookie': 12142138225658709835, 'match': {'in_port':
 26, 'dl_vlan': 1}, 'table_id': 2, 'table_group': 'evpl', 'priority': 20000, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_type': 'apply_actions', 'actions': [{'a
ction_type': 'pop_int'}, {'action_type': 'pop_vlan'}, {'action_type': 'output', 'port': 22}]}]}, {'owner': 'telemetry_int', 'cookie': 12175442460871458118, 'match': {'in_port': 22, 'dl_
vlan': 2222, 'dl_type': 2048, 'nw_proto': 6}, 'table_id': 0, 'table_group': 'evpl', 'priority': 20100, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_type': 'apply
_actions', 'actions': [{'action_type': 'push_int'}]}, {'instruction_type': 'goto_table', 'table_id': 2}]}, {'owner': 'telemetry_int', 'cookie': 12175442460871458118, 'match': {'in_port'
: 22, 'dl_vlan': 2222, 'dl_type': 2048, 'nw_proto': 17}, 'table_id': 0, 'table_group': 'evpl', 'priority': 20100, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_ty
pe': 'apply_actions', 'actions': [{'action_type': 'push_int'}]}, {'instruction_type': 'goto_table', 'table_id': 2}]}, {'owner': 'telemetry_int', 'cookie': 12175442460871458118, 'match':
 {'in_port': 22, 'dl_vlan': 2222}, 'table_id': 2, 'table_group': 'evpl', 'priority': 20000, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_type': 'apply_actions', 
'actions': [{'action_type': 'add_int_metadata'}, {'action_type': 'push_vlan', 'tag_type': 's'}, {'action_type': 'set_vlan', 'vlan_id': 2}, {'action_type': 'output', 'port': 11}]}]}, {'o
wner': 'telemetry_int', 'cookie': 12175442460871458118, 'match': {'in_port': 11, 'dl_vlan': 2, 'dl_type': 2048, 'nw_proto': 6}, 'table_id': 0, 'table_group': 'evpl', 'priority': 20100, 
'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_type': 'apply_actions', 'actions': [{'action_type': 'add_int_metadata'}, {'action_type': 'output', 'port': 25}]}]}, 
{'owner': 'telemetry_int', 'cookie': 12175442460871458118, 'match': {'in_port': 11, 'dl_vlan': 2, 'dl_type': 2048, 'nw_proto': 17}, 'table_id': 0, 'table_group': 'evpl', 'priority': 201
00, 'idle_timeout': 0, 'hard_timeout': 0, 'instructions': [{'instruction_type': 'apply_actions', 'actions': [{'action_type': 'add_int_metadata'}, {'action_type': 'output', 'port': 25}]}
]}, {'owner': 'telemetry_int', 'cookie': 12175442460871458118, 'match': {'in_port': 26, 'dl_vlan': 2}, 'table_id': 0, 'table_group': 'evpl', 'priority': 20000, 'idle_timeout': 0, 'hard_
timeout': 0, 'instructions': [{'instruction_type': 'apply_actions', 'actions': [{'action_type': 'send_report'}]}, {'instruction_type': 'goto_table', 'table_id': 2}]}, {'owner': 'telemet
ry_int', 'cookie': 12175442460871458118, 'match': {'in_port': 26, 'dl_vlan': 2}, 'table_id': 2, 'table_group': 'evpl', 'priority': 20000, 'idle_timeout': 0, 'hard_timeout': 0, 'instruct
ions': [{'instruction_type': 'apply_actions', 'actions': [{'action_type': 'pop_int'}, {'action_type': 'pop_vlan'}, {'action_type': 'output', 'port': 22}]}]}]
2023-11-13 12:41:42,633 - INFO [uvicorn.access] [httptools_impl.py:506:send] (MainThread) 127.0.0.1:51900 - "POST /api/kytos/mef_eline/v2/evc/metadata HTTP/1.1" 201
kytos $>                                                                                                                                                                                 

```

- Observed iperf3 data transfer after no network convergence was going on too:

```
root@int02:/home/amlight# iperf3 -c 10.22.21.3 -i 5 -t 60
Connecting to host 10.22.21.3, port 5201
[  4] local 10.22.21.2 port 56440 connected to 10.22.21.3 port 5201
[ ID] Interval           Transfer     Bandwidth       Retr  Cwnd
[  4]   0.00-5.00   sec  11.6 GBytes  20.0 Gbits/sec  1387    877 KBytes       
[  4]   5.00-10.00  sec  12.6 GBytes  21.7 Gbits/sec  796    833 KBytes       
[  4]  10.00-15.00  sec  11.8 GBytes  20.2 Gbits/sec  671    867 KBytes       
[  4]  15.00-20.00  sec  11.4 GBytes  19.5 Gbits/sec  647    912 KBytes       
[  4]  20.00-25.00  sec  11.5 GBytes  19.8 Gbits/sec  592    788 KBytes       
[  4]  25.00-30.00  sec  13.1 GBytes  22.5 Gbits/sec  901    588 KBytes       
[  4]  30.00-35.00  sec  11.4 GBytes  19.6 Gbits/sec  656    629 KBytes       
[  4]  35.00-40.00  sec  11.2 GBytes  19.2 Gbits/sec  598    645 KBytes       
[  4]  40.00-45.00  sec  12.5 GBytes  21.4 Gbits/sec  765    506 KBytes       
[  4]  45.00-50.00  sec  11.5 GBytes  19.7 Gbits/sec  674    547 KBytes       
[  4]  50.00-55.00  sec  11.2 GBytes  19.3 Gbits/sec  746    676 KBytes       
[  4]  55.00-60.00  sec  12.8 GBytes  21.9 Gbits/sec  775    983 KBytes       
- - - - - - - - - - - - - - - - - - - - - - - - -
[ ID] Interval           Transfer     Bandwidth       Retr
[  4]   0.00-60.00  sec   142 GBytes  20.4 Gbits/sec  9208             sender
[  4]   0.00-60.00  sec   142 GBytes  20.4 Gbits/sec                  receiver

```

### End-to-End Tests

N/A yet

### CODE REVIEW

File: managers/int.py [hunk 1]
```diff
@@ -31,6 +37,162 @@ def __init__(self, controller: Controller) -> None:
...
+            pp = self.srcs_pp.get(link.endpoint_b.id)
+        if not pp or not pp.evc_ids:
+            return
+
+        # This sleep is to wait for at least a few seconds to ensure that the other
+        # proxy port would also have been considered up since the request
+        # on topology setting metadata might end up delaying, check out issue
+        # https://github.com/kytos-ng/of_lldp/issues/100
+        # TODO this will be optimized later on before releasing this NApp
+        await asyncio.sleep(5)
```
Comment (italovalcy) [inline managers/int.py:133]: I didnt understand this wait here. Can you please elaborate more about it?

**Review by italovalcy — APPROVED** [review]
Great work done so far on the INT Telemetry Napp, @viniarck. It looks good to me, I just left one comment to get more clear idea about the goal of the sleep call.

File: managers/int.py (hunk 1 above)
Comment (viniarck) [reply managers/int.py:133]: @italovalcy, this sleep will get removed once https://github.com/kytos-ng/of_lldp/issues/100, is addressed. But, yes, let me elaborate on it:

To make sure that indeed [both proxy ports are UP - for bidirectional EVCs - ](https://github.com/kytos-ng/telemetry_int/blob/feat/handle_proxy_port_link/managers/int.py#L163-L164) we're also relying on the `loop` state that's set via metadata, which `of_lldp` will set via a request, so there's a possibility that as highlighted in the linked issue, that this request is being made on a [of_lldp handler](https://github.com/kytos-ng/of_lldp/blob/master/main.py#L186-L192), which will get executed in the `app` threadpool at the moment, then that thread might get delayed in a extreme worst case scenario, and then maybe the metadata won't be there yet, so proxy ports wouldn't be considered up, and consequently not safe to assume that flows could get installed. In that issue, `of_lldp` will set the metadata upfront in memory, eliminating this wait here to ensure the request has suceeded, so this sleep is also assuming that 5 secs would be a worst case in the mean time, which should be OK, but far from ideal, that's that rationale.

**Comment by viniarck** [comment]
@italovalcy, thanks for you review, let me know if you have any other follow up questions.

File: managers/int.py (hunk 1 above)
Comment (viniarck) [reply managers/int.py:133]: btw, there's also another one that's desirable to have when it comes to enhancing loop detection robustness, currently they're susceptible to starvation, this part it'll be handled on another issue https://github.com/kytos-ng/of_lldp/issues/97

**Pull request merged**